x/stakeibc/keeper/unbonding_records_get_host_zone_unbondings_msgs_test.go

```diff
@@ -594,6 +594,139 @@ func (s *KeeperTestSuite) TestGetTotalUnbondAmount() {
 	s.Require().Zero(s.App.StakeibcKeeper.GetTotalUnbondAmount(s.Ctx, emptyUnbondings).Int64())
 }
 
+func (s *KeeperTestSuite) TestRefreshUserRedemptionRecordNativeAmounts() {
+	// Define the expected redemption records after the function is called
+	redemptionRate := sdk.MustNewDecFromStr("1.999")
+	expectedUserRedemptionRecords := []recordtypes.UserRedemptionRecord{
+		// StTokenAmount: 1000 * 1.999 = 1999 Native
+		{Id: "A", StTokenAmount: sdkmath.NewInt(1000), Amount: sdkmath.NewInt(1999)},
+		// StTokenAmount: 999 * 1.999 = 1997.001, Rounded down to 1997 Native
+		{Id: "B", StTokenAmount: sdkmath.NewInt(999), Amount: sdkmath.NewInt(1997)},
+		// StTokenAmount: 100 * 1.999 = 199.9, Rounded up to 200 Native
+		{Id: "C", StTokenAmount: sdkmath.NewInt(100), Amount: sdkmath.NewInt(200)},
+	}
+	expectedTotalNativeAmount := sdkmath.NewInt(1999 + 1997 + 200)
+
+	// Create the initial records which do not have the end native amount
+	for _, expectedUserRedemptionRecord := range expectedUserRedemptionRecords {
+		initialUserRedemptionRecord := expectedUserRedemptionRecord
+		initialUserRedemptionRecord.Amount = sdkmath.ZeroInt()
+		s.App.RecordsKeeper.SetUserRedemptionRecord(s.Ctx, initialUserRedemptionRecord)
+	}
+
+	// Call the refresh user redemption record function
+	// Note: an extra ID ("D"), is passed into this function that will be ignored
+	// since there's not user redemption record for "D"
+	redemptionRecordIds := []string{"A", "B", "C", "D"}
+	actualTotalNativeAmount := s.App.StakeibcKeeper.RefreshUserRedemptionRecordNativeAmounts(
+		s.Ctx,
+		HostChainId,
+		redemptionRecordIds,
+		redemptionRate,
+	)
+
+	// Confirm the summation is correct and the user redemption records were updated
+	s.Require().Equal(expectedTotalNativeAmount.Int64(), actualTotalNativeAmount.Int64(), "total native amount")
+	for _, expectedRecord := range expectedUserRedemptionRecords {
+		actualRecord, found := s.App.RecordsKeeper.GetUserRedemptionRecord(s.Ctx, expectedRecord.Id)
+		s.Require().True(found, "record %s should have been found", expectedRecord.Id)
+		s.Require().Equal(expectedRecord.Amount.Int64(), actualRecord.Amount.Int64(),
+			"record %s native amount", expectedRecord.Id)
+	}
+}
+
+// Tests RefreshUnbondingNativeTokenAmounts which indirectly tests
+// RefreshHostZoneUnbondingNativeTokenAmount and RefreshUserRedemptionRecordNativeAmounts
+func (s *KeeperTestSuite) TestRefreshUnbondingNativeTokenAmounts() {
```

Comment:
Do we need to test the case where there is a record value 0?

Reply:
I don't think so cause it will still just do the same multiplication